crates/wysiwyg/src/dom/range.rs

```diff
@@ -310,10 +328,27 @@ impl Range {
             }
 
             shared_path.push(min_leaf_path[i]);
+
+            if !allow_covered_nodes {
+                let location = self
+                    .find_location(&DomHandle::from_raw(shared_path.clone()));
+
+                if location.is_covered() {
+                    shared_path.pop();
+                    break;
+                }
+            }
         }
 
         DomHandle::from_raw(shared_path)
     }
+
+    fn find_location(&self, node_handle: &DomHandle) -> &DomLocation {
+        self.locations
+            .iter()
+            .find(|l| *l.node_handle.raw() == *node_handle.raw())
+            .unwrap()
+    }
```

Comment:
It would probably be safer to return an `Option<&DomLocation>` instead and handle the `None` case in the calling function. Also, this is done a few times in the codebase, so having it as a `pub(crate)  fn` so we can replace those usages could be useful.

Reply:
[`28e06db`](https://github.com/matrix-org/matrix-rich-text-editor/pull/423/commits/28e06dbf41aae9bb20763ee8abdbfb409259bfeb)